src/base/emu-i386/kvm.c

```diff
@@ -1053,6 +1125,7 @@ static unsigned int kvm_run(void)
 	 the kernel with KVM_RUN. The kernel side will first finish
 	 incomplete operations and then check for pending signals." */
       run->immediate_exit = 1;
+      kvm_compat_set_immediate_exit(1);
```

Comment:
Then at least please move the
`run->immediate_exit = 1;` inside, in the
`#else` block.

Reply:
done